libr/core/cfile.c

```diff
@@ -907,6 +907,14 @@ R_API RCoreFile *r_core_file_open(RCore *r, const char *file, int flags, ut64 lo
 			swstep = false;
 		}
 		r_config_set_i (r->config, "dbg.swstep", swstep);
+		// Set the correct debug handle
+		if (fd->plugin && fd->plugin->isdbg) {
+			const char *dh = r_str_ndup (file, (strstr (file, "://") - file));
```

Comment:
if it's freed, it should not be `const char *`, I think.

Reply:
what if there is no `://` in `file`?

Reply:
>if it's freed, it should not be const char *, I think.
Seems logical since you want to treat it as const, this is common practice with strdup but you can't control it's usage of malloc. I could explicitly cast the pointer down to char* when freeing or just remove const since it doesn't really matter.

>what if there is no :// in file?
Nothing is set since, as far as I know, it's not supposed to happen. Is there a valid uri that would set fd->plugin->isdbg to true without having ://? radare2.c does the same with -d

Reply:
drop the const